text/3371-cargo-target-directories.md

```diff
@@ -254,7 +293,8 @@ Both express the needs for either remapping or a global target directory that is
 - What related issues do you consider out of scope for this RFC that could be addressed in the future independently of the solution that comes out of this RFC?
 -->
 
-None I can think of.
+- Should we use a hash-based solution or simply a directory-named based one ? `bazel` using hashes indicates this solution is viable for them, it probably would be for `cargo` too and if we use both the hash and the directory name, it would stay fairly human-readable but this solution also has disadvantages.
+- Do we want to differentiate according to users ? `bazel` is a generic build tool, whereas `cargo` is not, so maybe differentiating on users is not necessary for us ?
 
 # Future possibilities
```

Comment:
What would be the path for making a central target directory the default?

Reply:
Heavy (and I mean **heavy**) advertising of the option and of `cargo metadata`, making the latter pretty much mandatory to interact with the cargo target directory in any way. Once all tools and people have been using it for some time, then we could envisage changing to a central target directory as the default but that will not be any time soon and will bring its slew of new problems: what defaults, on which platform, what's the upgrade path for people not using anything of this in their local CI and not breaking their (perfectly valid) use case and so on.

I honestly don't think we'll ever be able to make this the default, though I would be pleasantly surprised if we manage it

Reply:
I think changing the default has a lot of potential and would like to see a more thorough look at how we could do this in Future Possibilities (what could it negatively impact, how we could get around that, etc).

Reply:
done